portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,26 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+    messages.add("Test 1");
+    messages.add("Test 2");
+    messages.add("Test 3");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    String json = convertToJsonUsingGson(messages);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJsonUsingGson(List<String> data) {
```

Comment:
You dont actually need this method since its actually not saving you any lines. Its good to pull out concepts when there a self contained bit of code you want to reuse, or when the parent function is getting long. But also, if you compare this to what it would be like without it, the calling function remains the same length.

    Gson gson = new Gson()
    response.setContentType("application/json;");
    response.getWriter().println(gson.toJson(messages);

Reply:
That said, its not *bad* to have it, so if it makes more sense to you organizationally, its perfectly fine too